regression/snapshot-harness/pointer-to-array-int/main.c

```diff
@@ -0,0 +1,42 @@
+#include <assert.h>
+#include <malloc.h>
+
+int *first;
+int *second;
+int *third;
+int array_size;
+const int *prefix;
+int prefix_size;
+
+void initialize()
+{
+  first = (int *)malloc(sizeof(int) * 5);
+  first[0] = 0;
+  first[1] = 1;
+  first[2] = 2;
+  first[3] = 3;
+  first[4] = 4;
+  second = first;
+  array_size = 5;
+  third = &array_size;
+}
+
+void checkpoint()
+{
+}
+
+int main()
+{
+  initialize();
+  checkpoint();
+
+  assert(first == second);
+  /* assert(array_size >= prefix_size); */
```

Comment:
Ditto with these asserts - worth explaining why they don't currently hold/when they might be able to be enabled.

Reply:
This one was postponed until the improved support for dynamic allocation will be implemented in #4578. I've added a comment.